pilot/pkg/model/context.go

```diff
@@ -200,17 +201,7 @@ type GeneratorMetadata struct {
 // Note: any errors returned will completely close the XDS stream. Use with caution; typically and empty
 // or no response is preferred.
 type XdsResourceGenerator interface {
-	Generate(proxy *Proxy, push *PushContext, w *WatchedResource, updates *PushRequest) (Resources, error)
-	Metadata() *GeneratorMetadata
-}
-
-// BaseGenerator is the base generator.
-type BaseGenerator struct{}
-
-var baseMetadata = &GeneratorMetadata{false}
-
-func (bg *BaseGenerator) Metadata() *GeneratorMetadata {
-	return baseMetadata
+	Generate(proxy *Proxy, push *PushContext, w *WatchedResource, updates *PushRequest) (Resources, *XdsLogDetails, error)
```

Comment:
do we need XdsLogDetails to be a pointer? A non pointer seems like it would work here fine, and the zero value is equivalent to `nil` (incremental=false, no additional info) while avoiding nilchecks and heap allocs

Reply:
Yes. We can avoid it. Changed PTAL